lib/src/command/outdated.dart

```diff
@@ -111,55 +122,83 @@ class OutdatedCommand extends PubCommand {
 
     Future<_PackageDetails> analyzeDependency(PackageRef packageRef) async {
       final name = packageRef.name;
-      final current = (entrypoint.lockFile?.packages ?? {})[name]?.version;
-      final source = packageRef.source;
-      final available = (await cache.source(source).doGetVersions(packageRef))
-          .map((id) => id.version)
-          .toList()
-            ..sort(argResults['pre-releases'] ? null : Version.prioritize);
-      final upgradable = upgradablePackages
-          .firstWhere((id) => id.name == name, orElse: () => null)
-          ?.version;
-      final resolvable = resolvablePackages
-          .firstWhere((id) => id.name == name, orElse: () => null)
-          ?.version;
-      final latest = available.last;
-      final description = packageRef.description;
+      final current = (entrypoint.lockFile?.packages ?? {})[name];
+
+      final upgradable = upgradablePackages.firstWhere((id) => id.name == name,
+          orElse: () => null);
+      final resolvable = resolvablePackages.firstWhere((id) => id.name == name,
+          orElse: () => null);
+
+      // Find the latest version, and if it's overridden
+      var latestIsOverridden = false;
+      // If present as a dependency of dev_dependency we use this ref
+      var latest = await _getLatest(rootPubspec.dependencies[name]?.toRef());
+      latest ??= await _getLatest(rootPubspec.devDependencies[name]?.toRef());
+      // If not overridden in current resolution we can use this ref
+      if (!entrypoint.root.pubspec.dependencyOverrides.containsKey(name)) {
+        latest ??= await _getLatest(current?.toRef());
+      }
+      // If not overridden and present in either upgradable or resolvable we
+      // use this reference to find the latest
+      if (!upgradablePubspec.dependencyOverrides.containsKey(name)) {
```

Comment:
The dependency overrides of upgradeblePubspec and resolvablePubspec  will always be either empty or the same as for entrypoint.root?

```
if (!entrypoint.root.pubspec.dependencyOverrides.containsKey(name)) {
  latest ??= await _getLatest(current?.toRef()) ?? await _getLatest(upgradable?.toRef()) ?? await _getLatest(resolvable?.toRef());
}
```

And we will always get a version if the ref is non-null (is this true??)

```
if (!entrypoint.root.pubspec.dependencyOverrides.containsKey(name)) {
  latest ??= await _getLatest((current ?? upgradable ?? resolvable)?.toRef());
}
```
And I guess we always have at least one of the candidates:
```
if (!entrypoint.root.pubspec.dependencyOverrides.containsKey(name)) {
  latest ??= await _getLatest((current ?? upgradable ?? resolvable).toRef());
}
```

Reply:
`current` can have overrides without `upgradable` and `resolvable` having overrides..

In any case, I figured it was preferable not to read the `argResult['dependency-overrides']` after creating the pubspecs for `upgradable` and `resolvable`.

Hence, I opted to derive whether or not it was overridden from the `(upgradable|resolvable)Pubspec` -- this has the benefit that if we change how overrides work in the future, or where they are applied it doesn't not affect this code.